torch/quantization/quantize_fx.py

```diff
@@ -41,20 +42,99 @@ def _fuse_fx(
+            # scope for this would be (module_path="sub", module_type=Sub)
+            return x.transpose(1, 2)
+
+    class M(torch.nn.Module):
+        def __init__(self):
+            self.sub = Sub()
+
+        def forward(self, x):
+            # This will be a call_method Node as well,
+            # scope for this would be (module_path="", None)
+            x = x.transpose(1, 2)
+            x = self.sub(x)
+            return x
+
+    """
+    def __init__(self, module_path: str, module_type: Any):
+        super().__init__()
+        self.module_path = module_path
+        self.module_type = module_type
+
+class ScopeContextManager(object):
+    """ A context manager to track the Scope of Node during symbolic
+    tracing.
+    When entering a forward function of a Module, we'll update the scope information of
+    the current module, and when we exit, we'll restore the previous scope information.
+    """
+    def __init__(
+            self,
+            scope: Scope,
+            current_module: torch.nn.Module,
+            current_module_path: str):
+        super().__init__()
+        self.prev_module_type = scope.module_type
+        self.prev_module_path = scope.module_path
+        self.scope = scope
+        self.scope.module_path = current_module_path
+        self.scope.module_type = type(current_module)
+
+    def __enter__(self):
+        return
+
+    def __exit__(self, *args):
+        self.scope.module_path = self.prev_module_path
+        self.scope.module_type = self.prev_module_type
+        return
+
+
+class QuantizationTracer(Tracer):
     def __init__(self, skipped_module_names: List[str],
                  skipped_module_classes: List[Callable]):
         super().__init__()
         self.skipped_module_names = skipped_module_names
         self.skipped_module_classes = skipped_module_classes
-
-    def is_leaf_module(self, m, module_qualified_name):
-        return (m.__module__.startswith('torch.nn') and
+        # NB: initialized the module_type of top level module to None
+        # we are assuming people won't configure the model with the type of top level
+        # module here, since people can use "" for global config
+        # We can change this if there is a use case that configures
+        # qconfig using top level module type
+        self.scope = Scope("", None)
+        self.node_name_to_scope : Dict[str, Tuple[str, Any]] = {}
+
+    def is_leaf_module(self, m: torch.nn.Module, module_qualified_name : str) -> bool:
+        return (m.__module__.startswith("torch.nn") and
                 not isinstance(m, torch.nn.Sequential)) or \
             module_qualified_name in self.skipped_module_names or \
             type(m) in self.skipped_module_classes or \
             isinstance(m, _FusedModule)
 
+    def call_module(self, m: torch.nn.Module, forward: Callable[..., Any], args : Tuple[Any, ...], kwargs : Dict[str, Any]) -> Any:
+        module_qualified_name = self.path_of_module(m)
```

Comment:
@jamesr66a this is copied from default `call_module` code, would it be a problem if the default code changes? do you think it's better to provide an API in `call_module` to allow this kind of extension?

Reply:
I think the future-proofed way to do this is to instantiate the guard and call `super().call_module`

Reply:
OK, sg